> MSVC reports "warning C4701: potentially uninitialized local variable
'host' used" in ngx_http_proxy_create_request(), which is treated as
an error due to the -WX compiler option:
> src\http\modules\ngx_http_proxy_module.c(1427) : warning C4701:
potentially uninitialized local variable 'host' used

we can safely omit such details


> The variable cannot actually be used uninitialized, since all uses of
host.data are guarded by a host.len check, but the compiler is unable
to prove this.  Fixed by initializing the whole variable.  

we can omit it too, NGX_SUPPRESS_WARN will self-document this for us

> The same
pattern in the gRPC and proxy v2 modules is updated for consistency.

actually, all three modules are suffered from false warnings,
I suggest to keep the log summary only and remove everything else

--- a/src/http/modules/ngx_http_grpc_module.c
+++ b/src/http/modules/ngx_http_grpc_module.c
@@ -811,6 +811,9 @@ ngx_http_grpc_create_request(ngx_http_request_t *r)
     /* :authority header */
 
     host.len = 0;
+#if (NGX_SUPPRESS_WARN)
+    host.data = NULL;
+#endif
 
     if (glcf->host_value
         && ngx_http_complex_value(r, glcf->host_value, &host) != NGX_OK)

--- a/src/http/modules/ngx_http_proxy_module.c
+++ b/src/http/modules/ngx_http_proxy_module.c
@@ -1226,6 +1226,9 @@ ngx_http_proxy_create_request(ngx_http_request_t *r)
     ctx = ngx_http_get_module_ctx(r, ngx_http_proxy_module);
 
     host.len = 0;
+#if (NGX_SUPPRESS_WARN)
+    host.data = NULL;
+#endif
 
     if (plcf->host_value
         && ngx_http_complex_value(r, plcf->host_value, &host) != NGX_OK)

--- a/src/http/modules/ngx_http_proxy_v2_module.c
+++ b/src/http/modules/ngx_http_proxy_v2_module.c
@@ -446,6 +446,9 @@ ngx_http_proxy_v2_create_request(ngx_http_request_t *r)
     /* :authority header */
 
     host.len = 0;
+#if (NGX_SUPPRESS_WARN)
+    host.data = NULL;
+#endif
 
     if (plcf->host_value
         && ngx_http_complex_value(r, plcf->host_value, &host) != NGX_OK)

